paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -492,21 +493,15 @@ def get_error_from_utilization(utilization, setpoint, current_instances):
         return 0.0
 
 
-def get_autoscaling_info(marathon_client, service, instance, cluster, soa_dir):
-    service_config = load_marathon_service_config(
-        service=service,
-        instance=instance,
-        cluster=cluster,
-        soa_dir=soa_dir,
-    )
+def get_autoscaling_info(marathon_clients, service_config):
     if service_config.get_max_instances() and service_config.get_desired_state() == 'start':
-        all_marathon_tasks, all_mesos_tasks = get_all_marathon_mesos_tasks(marathon_client)
+        apps_with_clients = get_marathon_apps_with_clients(marathon_clients, embed_tasks=True)
+        all_mesos_tasks = get_all_running_tasks()
```

Comment:
This will be so slow :( can you make sure we are calling the cached version of this?